storage/timestamp.js

```diff
@@ -0,0 +1,41 @@
+var redis;
+var locks = {};
+module.exports = function (core, config) {
+	redis = require('redis').createClient();
+	redis.select(config.redisDB);
+	core.on("edit", function(edit, callback) {
+		if (!edit.updateTime) edit.updateTime = new Date().getTime();
+		makeTimeUnique(edit, "updateTime", callback);
+	}, "validation");
+	core.on("text", function(text, callback) {
+		if (!text.time) text.time = new Date().getTime();
+		makeTimeUnique(text, "time", callback);
+	}, "validation");
+};
+
+
+function makeTimeUnique(action, propertyName, callback) {
```

Comment:
I will merge this. However, locks in JS will have to be rewritten when we make our application distributed, and it wastes memory and CPU.

Its better to use Redis features for this, e.g.

```javascript
var keyPrefix = 'timestamp:' + action.to + ':',
      expiry = 100; // number of milliseconds redis will remember this key.

redis.incr(keyPrefix + time, function(times) {
   // number of times we encountered that timestamp till now.
   if(times == 1) return;
   var newTime = timestamp + (times-1);
   action[propertyName] = newTime;
   redis.psetex(keyPrefix + newTime, 1, expiry);
});

redis.pexpire(keyPrefix + time, expiry);
```
The number of keys in Redis will be approximately equal to the number of messages that come in every 100 milliseconds. Your solution will have fewer keys if we add expiry to it (but probably more keys without expiry, due to our current pattern of inactive rooms)